### PR TITLE
Add spellcheck to CI

### PR DESCRIPTION
**Pull request by notgull**
winit is a large codebase and manually checking for typos is infeasible.
This commit adds a spellcheck hook using cspell to the CI. That way we
can be sure that there are no typos in our code before we commit.

----

WIP until I fix all of the typos.

- [ ] Tested on all platforms changed
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

File: src/platform/wayland.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 //!
 //! ## Client-side decorations
 //!
-//! Winit provides client-side decorations by default, but the behaviour can
```
Comment (kchibisov) [inline src/platform/wayland.rs:10]: That's a correct spelling btw.

File: src/platform/web.rs [hunk 2]
```diff
@@ -82,7 +82,7 @@ pub trait WindowExtWebSys {
     /// would prevent that.
     ///
     /// Some events are impossible to prevent. E.g. Firefox allows to access the native browser
-    /// context menu with Shift+Rightclick.
+    /// context menu with Shift+Right-click.
```
Comment (kchibisov) [inline src/platform/web.rs:85]: looks arbitrary.

File: src/platform/web.rs [hunk 3]
```diff
@@ -274,7 +274,7 @@ pub trait CustomCursorExtWebSys {
     /// but browser support for image formats is inconsistent. Using [PNG] is recommended.
     ///
     /// [PNG]: https://en.wikipedia.org/wiki/PNG
-    fn from_url(url: String, hotspot_x: u16, hotspot_y: u16) -> CustomCursorSource;
+    fn from_url(url: String, hot_spot_x: u16, hot_spot_y: u16) -> CustomCursorSource;
```
Comment (kchibisov) [inline src/platform/web.rs:277]: it's usually spelled like that(hotspot) in args.

File: src/platform/web.rs (hunk 2 above)
Comment (notgull) [reply src/platform/web.rs]: Generally for combination words (like Rightclick) I try to split it into two words unless it isn't possible (it's part of public API).

**Review by daxpedda — APPROVED** [review]
I've used [`cargo-spellcheck`](https://crates.io/crates/cargo-spellcheck) and [`typos`](https://crates.io/crates/typos) before, but this one looks good as well.

This is great!

**Comment by kchibisov** [comment]
Closing in favor of #3565

**Pull request closed without merging**